plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoRestCatalog.java

```diff
@@ -391,19 +428,55 @@ public void setViewPrincipal(ConnectorSession session, SchemaTableName schemaVie
     @Override
     public void dropView(ConnectorSession session, SchemaTableName schemaViewName)
     {
-        throw new TrinoException(NOT_SUPPORTED, "dropView is not supported for Iceberg REST catalog");
+        restSessionCatalog.dropView(convert(session), toIdentifier(schemaViewName));
     }
 
     @Override
     public Map<SchemaTableName, ConnectorViewDefinition> getViews(ConnectorSession session, Optional<String> namespace)
     {
-        return ImmutableMap.of();
+        SessionContext sessionContext = convert(session);
+        ImmutableMap.Builder<SchemaTableName, ConnectorViewDefinition> views = ImmutableMap.builder();
+        for (Namespace restNamespace : listNamespaces(session, namespace)) {
+            for (TableIdentifier restView : restSessionCatalog.listViews(sessionContext, restNamespace)) {
+                SchemaTableName schemaTableName = SchemaTableName.schemaTableName(restView.namespace().toString(), restView.name());
+                getView(session, schemaTableName).ifPresent(view -> views.put(schemaTableName, view));
+            }
+        }
+
+        return views.buildOrThrow();
     }
 
     @Override
     public Optional<ConnectorViewDefinition> getView(ConnectorSession session, SchemaTableName viewName)
     {
-        return Optional.empty();
+        return getIcebergView(session, viewName).flatMap(view -> {
+            SQLViewRepresentation sqlView = view.sqlFor(ICEBERG_VIEW_SQL_DIALECT);
+            if (!sqlView.dialect().equalsIgnoreCase(ICEBERG_VIEW_SQL_DIALECT)) {
+                throw new TrinoException(ICEBERG_UNSUPPORTED_VIEW_DIALECT, format("Cannot read unsupported dialect %s for view %s", sqlView.dialect(), viewName));
```

Comment:
Nit: use `.formatted()` instead of `format()`

```suggestion
                throw new TrinoException(ICEBERG_UNSUPPORTED_VIEW_DIALECT, "Cannot read unsupported dialect '%'s for view '%s'".formatted(sqlView.dialect(), viewName));
```